app/views/assays/_buttons.html.erb

```diff
@@ -1,4 +1,14 @@
-<% assay_word = item.is_modelling? ? t('assays.modelling_analysis') : t('assays.assay') %>
+<% assay_word ||=
+  if item.is_modelling?
+    t('assays.modelling_analysis')
+  elsif item.is_assay_stream?
+    t('assays.assay_stream')
+  elsif Seek::Config.isa_json_compliance_enabled && item.is_isa_json_compliant?
+    t('isa_assay')
+  else
+    t('assays.assay')
+  end
```

Comment:
I think this could be simplified to check `t("assays.#{item.assay_class.long_key}")` or something for the non-ISA assay cases

Reply:
It can but the button will say 'New __Experimental assay__ based on this one' instead of 'New __Assay__ based on this one'.